src/js/components/DSLInputField.js

```diff
@@ -72,7 +96,18 @@ class DSLInputField extends React.Component {
    * @param {SyntheticEvent} event - The change event
    */
   handleChange({target}) {
-    this.props.onChange(new DSLExpression(target.value));
+    this.setState({
+      expression: new DSLExpression(target.value)
```

Comment:
We always have one source of truth. The one that comes from the parent of the `DSLFilterField` component, and the one updated through the `onChange` callback. That's why the `componentWillReceiveProps` function always prefers props against state.

The problem is that if we want the UI to be editable until the time the timeout kicks-in, we must keep the current state of the expression somewhere. Keeping the `expression` instance instead of just the text value also allows us to update the error state of the field more interactively than waiting for the timeout.